### ISSUE TITLE
Does not seem to install or run, but no errors

### ISSUE THREAD

**Issue by e-t-l** [ALTaleX531/TranslucentFlyouts#90]
I downloaded [TF V3.0.0](https://github.com/ALTaleX531/TranslucentFlyouts/releases/tag/v3.0.0-insider-preview), unpacked the archive, and ran the installer bats as administrator. (Just to check, I manually opened an elevated CMD prompt and ran `start rundll32.exe "C:\Tools\TranslucentFlyouts.V3.x64\TFModern64.dll",Main /install` and `/start`. For good measure, I even rebooted then again manually ran `start rundll32.exe "TFModern64.dll",Main /start` (I did all the same steps with TFMain64.dll too), but there is no indication that it is running. I never see the console window for downloading symbols appear, and there are no visible changes to any flyouts. There is no "rundll32.exe" process active in Task Manager (I don't know if there should be or not). 

I also tried all the steps from issue #44: I did all the above with antivirus disabled, and afterward I rebooted the PC; in Windows settings, Transparency Effects are already enabled, and the Registry entry `HKLM\Microsoft\Windows\Dwm\ForceEffectMode` does not exist. Just to check, I tried setting `HKEY_CURRENT_USER\Software\TranslucentFlyouts\Menu\EnableCompatibilityMode` to `1`, but it made no difference.

I am running Windows 11 Enterprise x64, version 23H2, on a brand new Lenovo Legion 5i Pro (Gen 8).

Has anyone else had issues getting TF to work on Win 23H2?

**Comment by ALTaleX531** [comment]
I have updated the attachement, please see if it solve your problem.

**Comment by UnclWish** [comment]
Windows 10 Pro 22H2 19045.3996.
Immersive version installs and launchs without errors. Running launch_immersive.bat again says that service is already running. But no flayouts effects added.

**Comment by ALTaleX531** [comment]
> Windows 10 Pro 22H2 19045.3996. Immersive version installs and launchs without errors. Running launch_immersive.bat again says that service is already running. But no flayouts effects added.

@UnclWish It is currently intended for Windows 11, I will add support for Windows 10 in the official release.

**Comment by e-t-l** [comment]
> I have updated the attachement, please see if it solve your problem

I did all the same steps with the updated versions. Running the BATs as admin results in exit code 0 for each of them (which I believe means no errors) yet nothing happens. Even after rebooting, I never see symbols downloading

**Comment by ALTaleX531** [comment]
![image](https://github.com/ALTaleX531/TranslucentFlyouts/assets/105535231/bc9fb12e-fd0a-41b5-a225-2b3fed0e3a8f)
Cannot reproduce!

**Comment by e-t-l** [comment]
Hm. After launching TF, is there anything else I should see on a system level to indicate if it's running or not, e.g. a process in task manager?

**Comment by ALTaleX531** [comment]
You can find rundll32.exe in the task manager, or you can simply run the stop_win32.bat. If it is not running it will show a error message box to you.

**Comment by powerdoll** [comment]
same. i did step by step and doesn't seem to work

**Comment by e-t-l** [comment]
@powerdoll Ok now that there's 2 of us, maybe we can figure out what we have in common about our devices. 

One possible factor occurred to me that I haven't had time to check. Do you have any windows hardening settings in gpedit or Windows Security (for example, ASLR)?

**Comment by powerdoll** [comment]
> @powerdoll Ok now that there's 2 of us, maybe we can figure out what we have in common about our devices.
> 
> One possible factor occurred to me that I haven't had time to check. Do you have any windows hardening settings in gpedit or Windows Security (for example, ASLR)? I'm wondering, If TF works by patching the active memory, whether a security policy like memory address layout randomization could interfere with that.

nope, i actually have a fresh windows 11 install, just installed explorer blur mica, that's it

**Comment by e-t-l** [comment]
> nope, i actually have a fresh windows 11 install, just installed explorer blur mica, that's it

That's good to know! That actually rules out a lot. What edition to Windows do you have - Home, Pro, Enterprise?

(Also I just realized I accidentally closed this issue with my last comment. Its definitely NOT complete.)

**Comment by powerdoll** [comment]
Pro
yeah no worries, i dont know how ot make it work, i see like a mica effect inside the 7-zip program when right clicking but its not mica at all, it's weird, thats the only place it kinda "shows"

**Comment by powerdoll** [comment]
![Recording 2024-02-07 160250](https://github.com/ALTaleX531/TranslucentFlyouts/assets/105698999/dc85d241-e2b8-4bee-a6a4-b0ff579b7176)
you cant really appreciate it here but it shows mica inside 7-zip, and white anywhere else

**Comment by e-t-l** [comment]
Interesting. I was able to install MicaForEveryone just fine, which means the issue is not with the in-memory patching, like I was originally afraid might he the case with ASLR. I am still brainstorming how else to troubleshoot TF.

**Comment by e-t-l** [comment]
@ALTaleX531 is there a debug version of TF so I could get some sort of output or logs of what it's doing?

**Comment by ALTaleX531** [comment]
![image](https://github.com/ALTaleX531/TranslucentFlyouts/assets/105535231/be3d9a53-46f4-4809-907b-c4499fda5509)
This is a debug version and you can view the debug output via DbgView.
[Debug.zip](https://github.com/ALTaleX531/TranslucentFlyouts/files/14312040/Debug.zip)

**Comment by powerdoll** [comment]
i'm a linux user now, not my problem anymore >:D

**Comment by e-t-l** [comment]
> This is a debug version and you can view the debug output via DbgView.

Okay, this is the debug output I got: 
```
[12556] [TranslucentFlyouts.Win32] Framework::Startup rejected. [serviceRunning: false, processIsRundll32: true, processIsTFC: false]
[12556] [TranslucentFlyouts.Win32] ::Main invoked, [buildNumber: 22631]
```

Those are the only two lines produced no matter if I run it with the `/install`, `/start`, or `/stop` flags. I do not see any rundll32 instance being created in Task Manager. Also, running `/stop` when TF isn't already running does not produce any error. 

It almost seems like the flags are being ignored? I have no idea how that would happen or why. In your screenshot, you have highlighted a line that says "Application::StartService, user has confirmed, ready to call Application..." I never see anything like this. I only ever see the two lines I pasted above.

**Comment by e-t-l** [comment]
Hmm, I also just noticed that [DWMBlurGlass](https://github.com/Maplespe/DWMBlurGlass) doesn't work on my system (but it does work inside Windows Sandbox!). It says that it installs, but doesn't actually have any effect. And to my knowledge, it uses the same or similar transparency-modifying process as TF. On the other hand, [MicaForEveryone](https://github.com/MicaForEveryone/MicaForEveryone) works just fine for me. Weird.

@ALTaleX531 Could it have something to do with GPUs? The only other thing I can think of is that I have a Geforce RTX 4070 GPU. Do we know anything about discrete GPUs handling DWM differently?

**Comment by ALTaleX531** [comment]
Since that's really a debug version, you'll need to install the appropriate VC runtime libraries in the sandbox, and here's a [version](https://github.com/ALTaleX531/TranslucentFlyouts/files/14320059/Debug.zip) that removes that dependency and contains more information. I don't think it has anything to do with the GPU at all, from the information you provided TF seems to be experiencing a crash, which could also be what DwmBlurGlass is experiencing as they both use the same code.

**Comment by ALTaleX531** [comment]
@e-t-l  I think I've figured out why, because your cpu probably doesn't support the avx2 enhanced instruction set. I should be able to fix it this weekend.
@TechySC23 The code is just too complex and combining them together would require a lot of work. Maybe I'll think about that later.

**Comment by e-t-l** [comment]
> your cpu probably doesn't support the avx2 enhanced instruction set

I don't know what that it, but it seems unlikely that my CPU wouldn't support it. My pc is a brand new gaming PC with the latest-gen Intel i7 CPU. (And I had previously had TF running just fine on my *10-year-old* budget-bin netbook!)

But if you say you know how to fix it, then I'll trust that and eagerly await your update.

(FWIW, I have not yet had the opportunity to try out the version you shared in https://github.com/ALTaleX531/TranslucentFlyouts/issues/90#issuecomment-1950293337, but I've been trying to figure out if there's anything else weird about my system that would make TF not work. Can you confirm whether TF works normally on a Windows machine that has [Virtualization-Based Security (VBS)](https://learn.microsoft.com/en-us/windows-hardware/design/device-experiences/oem-vbs) enabled?

**Comment by ALTaleX531** [comment]
@e-t-l  Please check the latest version, theoretically TF is not affected by VBS.

**Comment by TechySC23** [comment]
> @e-t-l Please check the latest version, theoretically TF is not affected by VBS.

Even my CPU has VBS on still its working.

**Comment by e-t-l** [comment]
> here's a [version](https://github.com/ALTaleX531/TranslucentFlyouts/files/14320059/Debug.zip) that removes that dependency and contains more information. I don't think it has anything to do with the GPU at all, from the information you provided TF seems to be experiencing a crash

Even with that updated debug version, the output is still:
```
[34088] [TranslucentFlyouts.Win32] Framework::Startup rejected. [serviceRunning: false, processIsRundll32: true, processIsTFC: false]
[34088] [TranslucentFlyouts.Win32] ::Main invoked, [buildNumber: 22631]
```
There is no other output.

> @e-t-l Please check the latest version, theoretically TF is not affected by VBS.

I tried the latest version and, same as before, nothing happened when I ran the bat files on my host computer. And, same as before, when I ran it in Windows Sandbox, it did install and run successfully. I am really running out of ideas of why it would work in the Sandbox but not on the host PC. I have Avast Antivirus installed, but I have whitelisted all the files and folders for TF, *and* I have completely disabled Avast when I make these attempts. 

Please reopen this issue @ALTaleX531, it unfortunately is still not solved.

**Comment by ALTaleX531** [comment]
So if I'm not mistaken, the TF main service died suddenly, no specified function was called, no crash occurred as there was no indication that a crash dump file was generated, and the only possibility I can think of is that it was killed. I don't really know how to fix this.

**Comment by ALTaleX531** [comment]
If you are a pro user, you can try running the launch command in windbg and see if anything unusual happens.

**Comment by e-t-l** [comment]
> try running the launch command in windbg

Sorry, what do you mean by this? I've tried running all the batch file commands (`"TFmain64.dll",Main /install` and `"TFmain64.dll",Main /start`) while looking at WinDBG, and it only ever shows those two lines of output I've quoted in my previous messages. 

> If you are a pro user

I'm running Windows 11 Enterprise, if that's what you mean

**Comment by e-t-l** [comment]
Oh ok, got it. Sorry I had confused the names WinDBG and DbgView. Yes, I am a power user and I do have WinDBG but haven't tried it with TF yet. I'll do that and report back

**Comment by ALTaleX531** [comment]
It’s been quite a while. Any progress?

**Comment by ALTaleX531** [comment]
![image](https://github.com/ALTaleX531/TranslucentFlyouts/assets/105535231/d9fd31e4-5776-48c4-94fe-48a052616a46)
If you have UMCI enabled, then TF won't work because Rundll32 doesn't load it. Not sure if it helps you or not.

**Comment by e-t-l** [comment]
By the way, what exactly does the `/install` command do? I want to see if there's some part of the installation that's being blocked; it's possible I can set up the installation manually...

**Comment by ALTaleX531** [comment]
The /install command just makes itself start automatically after booting, if /start doesn't work then TF won't work either.

**Comment by e-t-l** [comment]
@ALTaleX531 I have made progress. I haven't figured out the root issue, but I've discovered that I can run TF from the `NT Authority\SYSTEM` account by using [PSEXEC](https://learn.microsoft.com/en-us/sysinternals/downloads/psexec). 

The installation/creation of a scheduled task doesn't seem to work from SYSTEM, though, so I'm figuring out the best way to do that manually. (Creating a scheduled task that runs as the SYSTEM user also doesn't appear to work, so I have to create it with admin privileges then call psexec.exe and pass along the start command)

**Comment by ALTaleX531** [comment]
Wait, the account you're currently using isn't an administrator account?

**Comment by e-t-l** [comment]
No, I am the Administrator account. In other words, I am a user account in the `Administrators` group. What I did is use PSEXEC (which requires Admin elevation to run anyway) in order to run commands from the `NT Authority\SYSTEM` userspace.

**Comment by ALTaleX531** [comment]
I understand what you're saying, the reason I'm asking is that all of TF's operations can be done with administrator privileges, which suggests that you may have made special changes to the PC or that TF's processes have been killed.

**Comment by e-t-l** [comment]
> which suggests that you may have made special changes to the PC or that TF's processes have been killed

I agree, it's very likely I made some change when originally setting up my PC that is causing Rundll32.exe to be killed. I still intend to investigate this further (I've been too busy with work recently to run TF with WinDBG). In the meantime, I have established a workaround, in case anyone else encounters a similar issue to me:

1. Install the [Microsoft Sysinternals Suite](https://apps.microsoft.com/detail/9p7knl5rwt25)
2. To check if this solution works for you, open an elevated CMD prompt and run: 
```
"%LocalAppData%\Microsoft\WindowsApps\PsExec.exe" -s -i Rundll32 "C:\path\to\TranslucentFlyouts.Win32.V3.x64\TFMain64.dll",Main /start
```
3. Assuming Step 2 successfully runs TF, create a batch file "LaunchTF.cmd" with the following contents:
```
taskkill /f /im "psexec.exe" >nul
start Rundll32 "C:\path\to\TranslucentFlyouts.Win32.V3.x64\TFMain64.dll",Main /start
start Rundll32 "C:\path\to\TranslucentFlyouts.Win32.V3.x86.optional\TFMain32.dll",Main /start
```
4. Create a Scheduled Task to run.... "only when user is logged on," "with highest privileges", with the Trigger "at log on," and the Action "Start a program" (Program/script: `"%LocalAppdata%\Microsoft\WindowsApps\PsExec.exe"`, Arguments: `-i -s cmd /c "C:\path\to\LaunchTF.cmd"`), with all Conditions unchecked, and Settings "stop the task if it runs longer than `15 seconds`."
